posts/app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,45 @@
+class PostsController < ApplicationController
+  def create
+    user = User.where(username: params[:username])
+    @post = Post.new(title: params[:title], content: params[:content], user: user.first)
+    if @post.save!
+      respond_to do |format|
+        format.json { render json: @post }
+      end
+    else
+      render nothing: true, status: 400
+    end
+  end
+
+  def index
+    @posts = Post.sorted
+    respond_to do |format|
+      format.json { render json: @posts }
+    end
+  end
+
+  def upvote
+    @post = Post.find(params[:id])
```

Comment:
É possível evitar a duplicidade?

Reply:
Evitar a duplicidade de Posts, sim. O ID é uma Primary Key em BD e portanto jamais se repete. Em uma situação ideal todos os IDs seriam UUIDs em vez de Integers, evitando então a enumeração de posts (por segurança).

Se estiver se referindo aos Votes, para evitar duplicidade seria necessário criar ao menos outras duas tabelas:
* Votes - Poderia ser Single-Table-Inheritance com Upvote e Downvote como opções/subclasses, ou um Enum indicando o tipo de Vote, ou o mesmo duas tabelas diferentes (Upvotes & Downvotes).
* UserVote - Para configurar a relação Many-to-many entre usuários e votes.

Isto aumentaria rapidamente a complexidade e tempo para a implementação, então quis focar em fazer uma versão mais simples e que funcionasse corretamente.